lib/pbench/server/database/models/datasets.py

```diff
@@ -1054,20 +1081,55 @@ def setvalue(
         "a.b.c" a value of "bar", and then query "a", you'll get back
         "a": {"b": {"c": "bar"}}}
 
+        We handle two keys specially:
+
+            dataset.name:   This is the primary resource name of a dataset, and
+                can be modified by the dataset's owner
+            server.deletion: This is the expected automatic deletion date of
+                the dataset, and can be modified so long as the new value is a
+                valid date and falls within the maximum server retention period
+                from the dataset's upload timestamp.
```

Comment:
My foremost concern is to prevent the documentation from misleading the reader.  Since the code uses symbolic references, it creates a risk if the surrounding comments try to specify the contents of the indirection.  In particular, when/if the values of the symbols are ever changed, how can we be sure that the documentation will be properly updated?  Is this a debt that we should take on?  If it's so important that the documentation mention the real value, then perhaps the code should use the real value, as well!  (I think we have very good reasons to avoid using literals in the code, and it seems a stretch (to me) to say that those reasons don't apply to the documentation; it's true that "bugs in the documentation" don't affect the operation of the code, but they can affect those who read the code...and, if we train the readers to ignore the documentation then we might as well omit it and save time and space.)